### PR TITLE
Allow updating Syncer internal annotations (for syncer view diffs) on status update

### PR DESCRIPTION
**Pull request by davidfestal**

### CODE REVIEW

File: staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/status_strategy.go [hunk 1]
```diff
@@ -53,6 +54,9 @@ func (a statusStrategy) GetResetFields() map[fieldpath.APIVersion]*fieldpath.Set
 	return fields
 }
 
+func isLocationDiffSyncerAnnotation(annotationKey string) bool {
```
Comment (sttts) [inline staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/status_strategy.go]: move into patch file

File: staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/status_strategy.go [hunk 2]
```diff
@@ -71,6 +83,21 @@ func (a statusStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.O
...
+			}
+		}
+	}
+	if annotations == nil && len(updatedInternalKCPAnnotations) > 0 {
+		annotations = make(map[string]string, len(updatedInternalKCPAnnotations))
+	}
+	for key, value := range updatedInternalKCPAnnotations {
+		annotations[key] = value
+	}
+	newCustomResourceObject.SetAnnotations(annotations)
```
Comment (sttts) [inline staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/status_strategy.go]: make it super visible that this is a patch

Comment (sttts) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/status_strategy.go]: same above

Comment (davidfestal) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/status_strategy.go]: Done in 5c1727bc7f07a2227c0c180b9c924156a7b72146

File: staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/status_strategy.go (hunk 1 above)
Comment (davidfestal) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/status_strategy.go]: Done in 5c1727bc7f07a2227c0c180b9c924156a7b72146

**Review by sttts — APPROVED** [review]

**Pull request merged**